python/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -170,7 +170,7 @@ def __assertPolicyResults(self, numTraitSets, context, traitSet=("entity",)):
         @param traitSet `List[str]` The set of traits to pass to
         the call to managementPolicy.
         """
-        traitSets = [traitSet for _ in range(numTraitSets)]
+        traitSets = [set(traitSet) for _ in range(numTraitSets)]
```

Comment:
[fix] Could we change `traitSet=("entity",)` to `traitSet={entity,}` instead of converting here? Then this conversion won't mask incorrect types fed into this method? Just we know people use tests as examples and so thinks like `L160` could be misleading (as they should be a set too).

Reply:
I did it this way because tuples are immutable, so are fine to use as default arguments, whereas if using a `set` then `pylint` says "Dangerous default value set() as argument (163:4) [dangerous-default-value]".

An alternative is to default to `None` then `if traitSet is None: traitSet = {"entity"}`, would that be preferred?

Reply:
Happy for a pylint disable (as it's definitely not dangerous in this case as manager isn't ever going to modify traitSet), or the latter, which ever you would prefer. I think the main thing is not using anything other than `set` at the call site, so people reading tests will have the 'setness' cemented in their brains.

Reply:
fwiw, I guess one thing I prefer about the default argument (given its safe here), is that its more readable what the behaviour is without having to look at the implementation.

Reply:
OK added the `set` default argument and suppressed pylint in 48b52532